File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/QuestionTranslationOutput.java
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import java.io.Serializable;
import java.math.*;
import java.time.*;
import java.util.*;
import jakarta.validation.constraints.*;

/**
 * QuestionTranslationOutput.
 */
@lombok.Getter @lombok.Setter
public  class QuestionTranslationOutput  implements Serializable {


    @NotNull
    private Long questionId ;

    
    private boolean required = true;

    
    private Integer rangeStart ;

    
    private Integer rangeEnd ;

    @NotNull @Size(max = 254)
    private String text ;

    @NotNull
    private QuestionType questionType ;

    
    private List<OptionTranslationOutput> options = new ArrayList<>();

    
    private boolean includeOther = false;









        public QuestionTranslationOutput addOptions(OptionTranslationOutput options) {
        this.options.add(options);
        return this;
        }


}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/QuestionTranslationOutput.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import jakarta.validation.constraints.*
import java.io.Serializable
import java.math.*
import java.time.*

/** QuestionTranslationOutput. */
data class QuestionTranslationOutput(
    @NotNull val questionId: Long? = null,
    val required: Boolean? = true,
    val rangeStart: Integer? = null,
    val rangeEnd: Integer? = null,
    @NotNull @Size(max = 254) val text: String? = null,
    @NotNull val questionType: QuestionType? = null,
    val options: MutableList<OptionTranslationOutput> = mutableListOf(),
    val includeOther: Boolean? = false,
) : Serializable {

    fun addOptions(options: OptionTranslationOutput): QuestionTranslationOutput {
        this.options += options
        return this
    }
}
